### PR TITLE
Added .NET 2.0 version (with LinqBridge)

### PR DESCRIPTION
**Pull request by bastianeicher**
I use ElfSharp in [Zero Install](https://github.com/0install/0install-win). To ensure it runs out-of-the-box without additional dependency installations on most Windows boxes it still targets .NET 2.0.

I added an additional project file to ElfSharp that targets .NET 2.0. In order to replace the missing LINQ functionality I used [LinqBridge](https://code.google.com/p/linqbridge/). The other compatibility changes were rather minor.

### CODE REVIEW

**Comment by konrad-kruczynski** [comment]
Hi,
sorry for the (veeery) late answer, but is the feature still necessary? I'm asking since, for instance, Mono has dropped the support for pre-4.0 profiles.

**Comment by bastianeicher** [comment]
Targeting .NET 2.0 has the nice side-effect that an application will run out-of-the-box on Windows Vista and newer.

The reason why I'm using an Elf-related library on Windows is because Zero Install's Publishing Tools allow you to create cross-platform packages. So I could be packaging a Linux binary on a Windows machine.

tl;dr: I still aim to support the huge install-base of not-quite-up-to-date Windows boxes out there.

**Comment by konrad-kruczynski** [comment]
I see. Then I generally don't see any problem in accepting the patch (although it would probably have to be updated), but since I do lack a development platform supporting .NET 2.0, I'll direct all problems regarding it to you ;) Is that OK with you?

**Comment by bastianeicher** [comment]
I've rebased the patch on your current master.
Sure, you can forward .NET 2.0 questions, that's fine by me.

File: ELFSharp/UImage/UImage.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 using System.Text;
 using System.Linq;
 using System.IO.Compression;
+using MiscUtil.IO;
```
Comment (konrad-kruczynski) [inline ELFSharp/UImage/UImage.cs]: Hey,
is this necessary?

**Comment by konrad-kruczynski** [comment]
One more thing. Could you add a license file for the LinqBridge?

**Comment by bastianeicher** [comment]
I removed the LinqBridge DLL from the commit and added a NuGet reference instead.
This way we're not distributing binaries over Git and the license information is embedded in the NuGet package.

File: ELFSharp/UImage/UImage.cs (hunk 1 above)
Comment (bastianeicher) [inline ELFSharp/UImage/UImage.cs]: This is here because `Stream` did not have the `CopyTo()` method in .NET 2.0. Instead I added it as an extension method in `MiscUtil.IO.Streams`.

Comment (konrad-kruczynski) [inline ELFSharp/UImage/UImage.cs]: I've eventually noticed and deleted a comment, but the email notification was already sent ;)

**Comment by konrad-kruczynski** [comment]
Yes, nuget is definitely the way to go.

**Comment by konrad-kruczynski** [comment]
Hello Bastian,
I'm writing because I plan to drop the support for the LinqBridge version of ELFSharp. It was not updated and it is not really maintained anymore - and I'm afraid I do not have time to do it. But I'd also like to know your opinion. Do you still use it? Are you interested in an up-to-date version targeted for .NET 2.0?

**Comment by bastianeicher** [comment]
Hello Konrad,

I still use ElfSharp in Zero Install which still targets .NET 2.0 to increase the number of Windows boxes it runs on out-of-the-box. However, I will retarget to a newer .NET version eventually. Also, I don't really need any new features in ElfSharp. So its fine by me if you drop support for .NET 2.0.

Regards
Bastian

**Comment by konrad-kruczynski** [comment]
Thanks for the quick answer!
